Add DAC support to STM32L1 series

tagunil opened the pull request:
Tested on a Nucleo-L152RE board. Both tests are passing, and the sample draws a nice sawtooth on my scope.
If that is good enough, I'll do the same for some more STM32 series.

tagunil committed:
soc: stm32l1: Enable DAC support

Enable STM32 DAC driver for STM32L1 series.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

tagunil committed:
boards: nucleo_l152re: Enable DAC support

Enable DAC support for Nucleo-L152RE board.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

tagunil committed:
tests: drivers: dac: Add Nucleo-L152RE board

Enable DAC tests for Nucleo-L152RE board.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

tagunil committed:
samples: drivers: dac: Add Nucleo-L152RE board

Enable DAC samples for Nucleo-L152RE board.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

gmarull commented on drivers/pinmux/stm32/pinmux_stm32l1x.h:
```diff
@@ -214,4 +214,9 @@
 #define STM32L1X_PINMUX_FUNC_PF10_ADC1_IN31  \
 	STM32_MODER_ANALOG_MODE
 
+#define STM32L1X_PINMUX_FUNC_PA4_DAC_OUT1 \
```
pinmux should be placed in order.

tagunil replied:
Looks like there are inconsistencies in pinmux ordering between STM32 series. I'll move these lines.

tagunil replied:
Done.

gmarull commented on boards/arm/nucleo_l152re/nucleo_l152re.dts:
```diff
@@ -39,6 +39,8 @@
 		sw0 = &user_button;
 		eeprom-0 = &eeprom;
+		adc1 = &adc1;
+		dac1 = &dac1;
```
I think this is redundant as sample does not force aliases. Could just use `DT_NODELABEL` as other boards do in the sample.

tagunil replied:
I see. Aliases were the old way of doing things. I'll clean this up for both Nucleo boards by a separate commit.

tagunil replied:
Done.

tagunil committed:
boards: nucleo_l073rz/l152re: use node labels in DAC samples and tests

Use node labels instead of aliases to select ADC and DAC instance
in samples and tests.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>

gmarull commented on boards/arm/nucleo_l152re/pinmux.c:
```diff
@@ -24,6 +24,9 @@ static const struct pin_config pinconf[] = {
 #if DT_HAS_NODE_STATUS_OKAY(DT_NODELABEL(adc1))
 	{STM32_PIN_PA0, STM32L1X_PINMUX_FUNC_PA0_ADC1_IN0},
 #endif
+#if DT_HAS_NODE_STATUS_OKAY(DT_NODELABEL(dac1))
```
Needs to be updated to `#if DT_NODE_HAS_STATUS(DT_NODELABEL(dac1), okay) && CONFIG_DAC`

tagunil replied:
Done, thanks!

erwango commented:
@tagunil can you rebase once again? I'm not sure how to to make buildkite happy
